### PR TITLE
adding support for mix autocomplete

### PR DESCRIPTION
**Pull request by davidhq**

### CODE REVIEW

File: mix-completion.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class MixCompletion < Formula
+  homepage "https://github.com/davidhq/mix-power-completion"
+  url "https://github.com/davidhq/mix-power-completion.git",
+    :revision => "e71d52e243f4c3a92c9b47788c3f283e453f43fe"
```
Comment (bfontaine) [inline mix-completion.rb]: Would you mind tagging a stable release here? Thanks!

Comment (davidhq) [inline mix-completion.rb]: how exactly? Head maybe? It's below.. I should move it up... just tell me if that's it please

Comment (bfontaine) [inline mix-completion.rb]: I see that it’s your repository. Run e.g. `git tag 1.0.0 && git push --tags` to tag a 1.0.0 version. Then  go on the “Releases” page of your repo [here](https://github.com/davidhq/mix-power-completion/releases). Copy the `.tar.gz` URL of your 1.0.0 release and use it here.
Don’t hesitate to ask any question if I’m unclear.

File: mix-completion.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class MixCompletion < Formula
+  homepage "https://github.com/davidhq/mix-power-completion"
+  url "https://github.com/davidhq/mix-power-completion/archive/0.8.0.tar.gz",
+    :revision => "e71d52e243f4c3a92c9b47788c3f283e453f43fe"
+  head "https://github.com/davidhq/mix-power-completion.git"
+  version "20150919"
```
Comment (davidhq) [inline mix-completion.rb]: I think I made it... I was looking at the `pip` formula and it didn't have .tar.gz release... sorry

is `version:` needed? Can I remove it next time? or should it be equal to tag for example (0.8.0)?

Comment (bfontaine) [inline mix-completion.rb]: Now that you use a `.tar.gz` link, you can remove both the `:revision` and the `version`. Homebrew will detect the version from the link.

**Comment by davidhq** [comment]
squashed the commits......

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by davidhq** [comment]
OH .. don't merge yet, sorry, I managed to mix up something with colors 

but do tell me about tests - do they pass?

**Comment by bfontaine** [comment]
> but do tell me about tests - do they pass?

I just asked our bot, we’re waiting for it :smiley:

**Comment by davidhq** [comment]
ok!

I fixed the colors

**Comment by bfontaine** [comment]
If you changed anything on your repo you’ll have to tag a new version and change it here.

File: mix-completion.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class MixCompletion < Formula
+  homepage "https://github.com/davidhq/mix-power-completion"
+  url "https://github.com/davidhq/mix-power-completion/archive/0.8.0.tar.gz"
```
Comment (bfontaine) [inline mix-completion.rb]: We also need a checksum below this line. Run `curl -sL https://github.com/davidhq/mix-power-completion/archive/0.8.0.tar.gz | shasum -a 256`. Paste then the checksum to have a line that looks like this:

```rb
  sha256 "a21e1...dummy...checksum...b741f01"
```

**Comment by davidhq** [comment]
done... 

I think force push works here

File: mix-completion.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class MixCompletion < Formula
+  homepage "https://github.com/davidhq/mix-power-completion"
+  url "https://github.com/davidhq/mix-power-completion/archive/0.8.1.tar.gz"
+  head "https://github.com/davidhq/mix-power-completion.git"
+  sha256 "948c44e32ca00500b67cbd6f0ee8aac601a5ac3af3c1d59d5afee3ed53b176ff"
```
Comment (bfontaine) [inline mix-completion.rb]: The `sha256` line should be just before the `head` one.

File: mix-completion.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+  def install
+    bash_completion.install "mix"
+  end
+
+  test do
+    assert_match "-F _mix",
+      shell_output("source #{bash_completion}/mix && complete -p mix")
+  end
+end
```
Comment (bfontaine) [inline mix-completion.rb]: The file should end with a newline.

File: mix-completion.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+class MixCompletion < Formula
+  homepage "https://github.com/davidhq/mix-power-completion"
```
Comment (bfontaine) [inline mix-completion.rb]: Please add a short description using `desc "…"` just above this line.

**Comment by bfontaine** [comment]
You can check that everything’s fine with `brew audit --strict mix-completion`

File: mix-completion.rb (hunk 5 above)
Comment (davidhq) [inline mix-completion.rb]: I fixed your other comments locally but regarding newline, are you sure?

screen shot 2015-09-21 at 23 31 46

**Comment by davidhq** [comment]
I pushed changes now but without newline at the end... it passes checks for me:

  1 file inspected, no offenses detected

let me know if it's ok now

File: mix-completion.rb (hunk 5 above)
Comment (bfontaine) [inline mix-completion.rb]: Yes, the trick is that it should end with a newline, but not an empty line; this is normally done by your editor.

Here is what happens if I open the formula in `vim` and save the file:
```diff
diff --git a/mix-completion.rb b/mix-completion.rb
index c1ed2cd..ede9af9 100644
--- a/mix-completion.rb
+++ b/mix-completion.rb
@@ -13,4 +13,4 @@ class MixCompletion < Formula
     assert_match "-F _mix",
       shell_output("source #{bash_completion}/mix && complete -p mix")
   end
-end
\ No newline at end of file
+end
```
Notice it added a newline but not an empty line.

Comment (davidhq) [inline mix-completion.rb]: So is it ok now?

There is newline in current pull request...

screen shot 2015-09-22 at 12 57 13

**Comment by bfontaine** [comment]
Merged; thank you @davidhq! :tada:

**Comment by davidhq** [comment]
yay! thank you for all the tips @bfontaine ....

**Pull request closed without merging**